src/database/mod.rs

```diff
@@ -840,6 +880,198 @@ pub trait DatabaseExt<'a>: Executor<'a, Database = Postgres> {
         let row_unprocessed = self.fetch_one(query_queued_deletion).await?;
         Ok(row_unprocessed.get::<bool, _>(0))
     }
+
+    async fn insert_new_batch_head(
+        self,
+        next_root: &Hash,
+        batch_type: BatchType,
+        commitments: &Commitments,
+        leaf_indexes: &LeafIndexes,
+    ) -> Result<(), Error> {
+        let query = sqlx::query(
+            r#"
+            INSERT INTO batches(
+                id,
+                next_root,
+                prev_root,
+                created_at,
+                batch_type,
+                commitments,
+                leaf_indexes
+            ) VALUES (DEFAULT, $1, NULL, CURRENT_TIMESTAMP, $2, $3, $4)
+            "#,
+        )
+        .bind(next_root)
+        .bind(batch_type)
+        .bind(commitments)
+        .bind(leaf_indexes);
+
+        self.execute(query).await?;
+        Ok(())
+    }
+
+    async fn insert_new_batch(
+        self,
+        next_root: &Hash,
+        prev_root: &Hash,
+        batch_type: BatchType,
+        commitments: &Commitments,
+        leaf_indexes: &LeafIndexes,
+    ) -> Result<(), Error> {
+        let query = sqlx::query(
+            r#"
+            INSERT INTO batches(
+                id,
+                next_root,
+                prev_root,
+                created_at,
+                batch_type,
+                commitments,
+                leaf_indexes
+            ) VALUES (DEFAULT, $1, $2, CURRENT_TIMESTAMP, $3, $4, $5)
+            "#,
+        )
+        .bind(next_root)
+        .bind(prev_root)
+        .bind(batch_type)
+        .bind(commitments)
+        .bind(leaf_indexes);
+
+        self.execute(query).await?;
+        Ok(())
+    }
+
+    async fn get_next_batch(self, prev_root: &Hash) -> Result<Option<BatchEntry>, Error> {
+        let res = sqlx::query_as::<_, BatchEntry>(
+            r#"
+            SELECT
+                id,
+                next_root,
+                prev_root,
+                created_at,
+                batch_type,
+                commitments,
+                leaf_indexes
+            FROM batches WHERE prev_root = $1
+            "#,
+        )
+        .bind(prev_root)
+        .fetch_optional(self)
+        .await?;
+
+        Ok(res)
+    }
+
+    async fn get_next_batch_without_transaction(self) -> Result<Option<BatchEntry>, Error> {
+        let res = sqlx::query_as::<_, BatchEntry>(
+            r#"
+            SELECT
+                batches.id as id,
+                batches.next_root as next_root,
+                batches.prev_root as prev_root,
+                batches.created_at as created_at,
```

Comment:
AFAIK sqlx handles fully qualified names, so you can leave this (and others) as `batches.created_at`

Reply:
I'm sure I have done it due to created_at conflict but will reassure myself. :)